scripts/rust_codegen.sh

```diff
@@ -0,0 +1,23 @@
+#! /usr/bin/env bash
+
+# Copyright 2023 SECO Mind Srl
+# SPDX-License-Identifier: Apache-2.0
+
+set -e pipefail
+
+PROTOC_MAJOR='24'
+# check only major version of protoc
+version=$(protoc --version | cut -d ' ' -f 2 | cut -d '.' -f 1)
+
+if [[ $version != "$PROTOC_MAJOR" ]]; then
+    echo "incompatible protoc version $version, expected $PROTOC_MAJOR" >&2
+    exit 1
+fi
+
+# Remove old code
```

Comment:
Is this if statement really necessarty? Shouldn't the file be substituted wen the last command (`cp`) is executed?